src/components/calcite-filter/calcite-filter.tsx

```diff
@@ -0,0 +1,93 @@
+import { Component, Element, Event, EventEmitter, Host, Prop, h } from "@stencil/core";
+import { debounce, forIn } from "lodash-es";
+
+const filterDebounceAmount = 250; // milliseconds
+
+@Component({
+  tag: "calcite-filter",
+  styleUrl: "calcite-filter.scss",
+  shadow: true
+})
+export class CalciteFilter {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Prop() textLabel: string;
+
+  @Prop() textPlaceholder: string;
+
+  @Prop() data: object[];
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calciteFilterChange: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  filter = debounce((value: string): void => {
+    const regex = new RegExp(value, "ig");
+    const find = (input, regex) => {
+      let found = false;
+      forIn(input, (value) => {
+        if (typeof value === "function") {
+          return;
+        }
+        if (Array.isArray(value) || (typeof value === "object" && value !== null)) {
+          if (find(value, regex)) {
+            found = true;
+          }
+        } else if (regex.test(value)) {
+          found = true;
+        }
+      });
+      return found;
+    };
+
+    const result = this.data.filter((item) => {
+      return find(item, regex);
+    });
+
+    this.calciteFilterChange.emit(result);
+  }, filterDebounceAmount);
+
+  inputHandler = (event: Event): void => {
+    const target = event.target as HTMLInputElement;
+    this.filter(target.value);
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  render() {
+    return (
+      <Host>
+        <label>
```

Comment:
Should we support a label/input as a slot or just an input as a slot?

Reply:
not sure. what's the use case you have in mind?

Reply:
We can tackle it as an enhancement if a good use case arises. Things like handling validation or setting input types.